### PR TITLE
Fix jax enable x64

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `doc/releases/changelog-dev.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] The PennyLane source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint pennylane/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
@dwierichs reported the following 
```
import jax
import pennylane as qml

x = jax.numpy.array(0.4, dtype=jax.numpy.float32)

@qml.qnode(qml.device("default.qubit", shots=10))
def node(x):
    qml.RX(x, 0)
    qml.measure(0)
    return qml.expval(qml.Z(0))

print(jax.config.jax_enable_x64)
>>> False # Expected because it was never activated
out = node(x)
print(jax.config.jax_enable_x64)
>>> True # Not expected.
```

**Description of the Change:**
`qml.compiler.active` first checks whether Catalyst is imported at all to avoid changing `jax_enable_x64` on module initialization.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-67969]

### CODE REVIEW

File: pennylane/compiler/compiler.py [hunk 1]
```diff
@@ -245,5 +246,6 @@ def circuit(phi, theta):
     >>> qml.qjit(circuit)(np.pi, np.pi / 2)
     -1.0
     """
-
+    if "catalyst" not in sys.modules:
+        return False
```
Comment (dime10) [inline pennylane/compiler/compiler.py:250]: Thanks @vincentmr ! Is there a way we can handle this generically in the `active_compiler` function above?

Something like (just for the idea):
```py
    for name, eps in AvailableCompilers.names_entrypoints.items():
        if name not in sys.modules:
            continue
        tracer_loader = eps["context"].load()
        if tracer_loader.is_tracing():
            return name
```

Comment (vincentmr) [reply pennylane/compiler/compiler.py:250]: Yes, that solution is better. I was going for a minimal change because it might have impact back in Catalyst, but I guess we could always revert in time for the release on Tuesday if needed. Let's try that.

Comment (dime10) [reply pennylane/compiler/compiler.py]: The only thing is I don't know what exactly "name" is, is it guaranteed to be the module name or can it be an arbitrary label of the compiler? 

I did a quick search and the `EntryPoint` class does have a `.module` attribute that contains the module part of the entry point (an entry in general looks like this: `importable.module:object.attr`).

But I'm also happy to approve as is if it works for catalyst, just thought I'd mention it if you wanted to confirm this does what we expect.

**Review by dime10 — APPROVED** [review]
👍

**Review by jay-selby — APPROVED** [review]

File: pennylane/compiler/compiler.py (hunk 1 above)
Comment (vincentmr) [reply pennylane/compiler/compiler.py]: > The only thing is I don't know what exactly "name" is, is it guaranteed to be the module name or can it be an arbitrary label of the compiler?

Same, but I think we're only looking for Catalyst in practice, for now. 

> But I'm also happy to approve as is if it works for catalyst, just thought I'd mention it if you wanted to confirm this does what we expect.

I checked that Catalyst `dynamic_one_shot` tests pass locally. Let's hope the CI goes green too.

**Review by albi3ro — APPROVED** [review]
